File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        Stack<TreeNode> st = new Stack<TreeNode>();
        TreeNode cur = root;
        List<Integer> res = new ArrayList<>();

        while (cur != null || !st.isEmpty()) {
            if (cur != null) {
                res.add(cur.val);
                st.push(cur.right);
                cur = cur.left;
            } else {
                cur = st.pop();
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-2.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st=[]
        res=[0]*len(temperatures)
        for i,t in enumerate(temperatures) :
            while st and t>temperatures[st[-1]]:
                idx=st.pop()
                res[idx]=i-idx
            st.append(i)
        return res



        